config.py

```diff
@@ -2504,25 +2512,44 @@ def make_epochs(
             overlap=rest_epochs_overlap,
             stop=stop)
         event_id = dict(rest=3000)
+        metadata = None
     else:  # Events for task runs
         if event_id is None:
             event_id = 'auto'
 
         events, event_id = mne.events_from_annotations(raw, event_id=event_id)
 
-    # Construct metadata from the epochs
-    if metadata_tmin is None:
-        metadata_tmin = tmin
+        # Construct metadata
+        #
+        # We only keep conditions that will be analyzed.
+        if isinstance(conditions, dict):
+            conditions = list(conditions.keys())
+        else:
+            conditions = list(conditions)  # Ensure we have a list
 
-    if metadata_tmax is None:
-        metadata_tmax = tmax
+        # TODO This function should be made public upstream
+        row_event_names = mne.utils.mixin._hid_match(
+            event_id=event_id, keys=conditions
+        )
```

Comment:
Any thoughts on that?

Reply:
we should avoid this. Can you do it eg using an EpochsArray just to get the indices?

Reply:
Maybe, but I'd like to avoid having to fiddle around too much … I think the cleanest approach would be to make `hid_match` public, or to duplicate it and include it in the Pipeline. WDYT?

Reply:
if would use here something like this for now:

```
import numpy as np
import mne

event_id = {'auditory/left': 1, 'auditory/right': 2, 'visual/left': 3,
            'visual/right': 4}

info = mne.create_info(['A'], 1)
data = np.random.randn(len(event_id), 1, 1)
events = np.zeros((len(event_id), 3), dtype=int)
events[:, 0] = events[:, 2] = np.arange(len(event_id)) + 1
epochs = mne.EpochsArray(data, info, event_id=event_id, events=events)
event_id_hid = epochs['left']
print(event_id_hid.event_id)
```